thrust/thrust/detail/pointer.h

```diff
@@ -134,84 +135,108 @@ class pointer : public thrust::detail::pointer_base<Element, Tag, Reference, Der
 public:
   using raw_pointer = typename super_t::base_type;
 
-  // constructors
+  _CCCL_HOST_DEVICE pointer()
```

Comment:
I believe we should replace all `_CCCL_HOST_DEVICE` with `_CCCL_API`, its shorter and does more things